plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/ResourceTaskContext.java

```diff
@@ -342,26 +350,21 @@ public void resolveResource(CancelIndicator cancelIndicator) {
 		parent.onDidRefreshContext(this, cancelIndicator);
 	}
 
-	/** Validate this context's main resource and send an issue update. */
-	public List<Issue> validateResource(CancelIndicator cancelIndicator) {
+	/**
+	 * Validate this context's main resource and send an issue update.
+	 */
+	public List<? extends LSPIssue> validateResource(CancelIndicator cancelIndicator) {
 		// validate
 		IResourceServiceProvider resourceServiceProvider = resourceServiceProviderRegistry
 				.getResourceServiceProvider(mainURI);
 		IResourceValidator resourceValidator = resourceServiceProvider.getResourceValidator();
-		List<Issue> issues = resourceValidator.validate(mainResource, CheckMode.ALL, cancelIndicator);
+		List<? extends LSPIssue> issues = LSPIssue
```

Comment:
TODO: respect comment one line below - validate may return null in presence of cancellation